data/src/split.py

```diff
@@ -17,6 +21,31 @@ def _generalized_check(script: str, word: str) -> bool:
     return bool(regex.match(regex_string, word))
 
 
+def _detect_best_script_name(
+    word: str, strict: bool = True
+) -> Union[Tuple[str, float], None]:
+    """Returns the most likely script name (rather than ISO 15924 code) the
+    word belongs to along with the corresponding confidence expressed as a
+    maximum likelihood estimate computed over the `word` sample. If `strict`
+    is enabled, then all the characters must belong to the same script and
+    `(None, None)` is returned on failure.
+
+    Example: "ژۇرنال" -> ("Arabic", 1.0).
+    """
+    script_counts: Dict[str, float] = collections.defaultdict(float)
+    for char in word:
+        script_counts[unicodedataplus.script(char)] += 1.0
+    script_probs = [(s, script_counts[s] / len(word)) for s in script_counts]
+    script_probs = sorted(script_probs, key=lambda x: x[1], reverse=True)
```

Comment:
`lambda x: x[1]` is slow compared to `operator.itemgetter(1)`.

Also, sort in-place with `.sort()` instead of making a copy. (I think it uses quicksort instead of merge-sort then?)